lib/chain.js

```diff
@@ -31,8 +31,8 @@ module.exports = (api) => {
     }
 
     filter(fn) {
-      this._update((array, resolve) => {
-        api.metasync.filter(array, fn, result => resolve(result));
+      this._update((array, resolve, reject) => {
```

Comment:
You may do that in this PR or another one, I don't feel strongly for either option. Keep in mind though that it'll need a new round of review if you decide to update this PR.